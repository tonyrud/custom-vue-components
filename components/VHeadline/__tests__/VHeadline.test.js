// import TestHelpers from '@@/.config/jest/TestHelpers'
// import Headline from './../VHeadline'

// const propsData = {
//   upper: false,
//   level: '2',
//   subHeader: true,
// }

// let testHelpers

describe('Headline', () => {
  it('runs', () => {
    expect(true).toBe(true)
  })
  // beforeEach(() => {
  //   testHelpers = new TestHelpers(Headline, {
  //     propsData,
  //   })
  // })

  // describe('Props', () => {
  //   it(':upper should be set to false', () => {
  //     expect(testHelpers.wrapper.props().upper).toBeFalsy()
  //   })

  //   it(':subHeader prop should have correct text', () => {
  //     expect(testHelpers.wrapper.props().subHeader).toBeTruthy()
  //   })

  //   it(':level prop is correct', () => {
  //     expect(testHelpers.wrapper.props().level).toEqual('2')
  //   })
  // })

  // describe('Markup and class changes', () => {
  //   it(':upper sets correct classes', () => {
  //     const headlineEl = testHelpers.wrapper.find('.heading__text')
  //     expect(headlineEl.classes()).not.toContain('upper')
  //     testHelpers.wrapper.setProps({ upper: true })
  //     expect(headlineEl.classes()).toContain('upper')
  //   })

  //   it(':level changes classes', () => {
  //     const headlineEl = testHelpers.wrapper.find('.heading__text')
  //     expect(headlineEl.classes()).toContain('h2')
  //     testHelpers.wrapper.setProps({ level: '3' })
  //     expect(headlineEl.classes()).not.toContain('h1')
  //     expect(headlineEl.classes()).toContain('h3')
  //   })
  // })

  // describe('Slots', () => {
  //   beforeEach(() => {
  //     testHelpers = new TestHelpers(Headline, {
  //       slots: {
  //         default: 'slot text',
  //       },
  //     })
  //   })

  //   it('slot renders text', () => {
  //     const slot = testHelpers.wrapper.find('.heading__text')
  //     expect(slot.text()).toBe('slot text')
  //   })
  // })
})
