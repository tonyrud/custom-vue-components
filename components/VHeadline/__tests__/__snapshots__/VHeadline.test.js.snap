// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline Markup and class changes matches snapshot 1`] = `"<span class=\\"heading__text h2 subheader\\"></span>"`;
